File: dtc-dist/src/DtcConf.java
package com.dtc.analytics.common;

import org.apache.hadoop.conf.Configuration;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.*;


/**
 * @Author :lihao
 * @Date :Created in 14:50 2019-04-28
 */
public class DtcConf {
    private static Logger logger = LoggerFactory.getLogger(DtcConf.class);
    private static Configuration conf = new Configuration();
    private final static String DTC_FILE = "dtc-mr.xml";


    /**
     * Initial or refresh configuration & constants. Before you starting
     * applications on Hadoop/HBase, you should call this method at first. If you
     * are testing units, calling this method might not be necessary.
     */
    public static void setup() {
        InputStream inputStream = null;
        try {
            inputStream = ClassLoader.getSystemResourceAsStream(DTC_FILE);
            //local test
//      inputStream = new FileInputStream(new File("/Users/lixuewei/workspace/private/dtcanalytics/conf/dtc-mr.xml"));
            if (null != inputStream) {
                conf.addResource(inputStream);
            } else {
                throw new IOException("Configure file not found: razor-mr.xml");
            }
        } catch (FileNotFoundException e) {
            logger.error("An error was caught", e);
        } catch (IOException e) {
            logger.error("An error was caught", e);
        } catch (Exception e) {
            logger.error("An error was caught", e);
        } finally {
            try {
                if (inputStream != null) {
                    inputStream.close();
                }
            } catch (Exception e) {
                logger.error("An error was caught", e);
            }
        }
    }


    public static Configuration getConf() {
        return conf;
    }


}

File: dtc-flink/src/main/scala/com/dtc/analytic/scala/common/DtcConf.scala
package com.dtc.analytic.scala.common

import java.io.{File, FileInputStream, FileNotFoundException, IOException, InputStream}
import java.nio.file.Path
import org.slf4j.{Logger, LoggerFactory}
import org.apache.hadoop.conf.Configuration

/**
  * Created on 2019-05-27
  *
  * @author :hao.li
  */
object DtcConf {

  def logger: Logger = LoggerFactory.getLogger(DtcConf.getClass)

  private val conf = new Configuration()

  def setup(): Unit = {
    var inputStream: InputStream = null
    try {
//      var inputStream = new FileInputStream(new File("conf/dtc-flink.xml"))
      var inputStream = DtcConf.getClass.getClassLoader.getResourceAsStream("dtc-flink.xml")
      if (inputStream != null) {
        conf.addResource(inputStream)
      } else {
        throw new IOException("Configure file not found: dtc-flink.xml")
      }
    } catch {
      case ex: FileNotFoundException => {
        logger.error("Configure file not found: dtc-flink.xml,and the cause is {}.", ex)
      }
      case ex: IOException => {
        logger.error("Configure file not found: dtc-flink.xml,and the cause is {}.", ex)
      }

    } finally {
      if (inputStream != null) {
        inputStream.close()
      }
    }

  }

  def getConf(): Configuration = {
    return conf
  }

}
